specs/signature-envelope-cose.md

```diff
@@ -154,8 +154,7 @@ The Notary Project signature supports the following unprotected header parameter
 Note: `<<` and `>>` are used to notate the CBOR byte string resulting from encoding the data item.
 
 - **[`x5chain`](https://datatracker.ietf.org/doc/html/draft-ietf-cose-x509-08#section-2)** (*array of bstr*): This REQUIRED parameter (label `33` by [IANA](https://www.iana.org/assignments/cose/cose.xhtml#header-parameters)) contains the ordered list of X.509 certificate or certificate chain ([RFC5280](https://datatracker.ietf.org/doc/html/rfc5280)) corresponding to the key used to digitally sign the COSE. The certificate chain is represented as an array of certificate, each certificate in the array is DER encoded and then wrapped in a CBOR byte string. The certificate containing the public key corresponding to the key used to digitally sign the COSE MUST be the first certificate, followed by the intermediate and root certificates in the correct order. Refer [*Certificate Chain* unsigned attribute](signature-specification.md#unsigned-attributes) for more details. Optionally, this header can be presented in the protected header.
-- **`io.cncf.notary.timestampSignature`** (*bstr*): This OPTIONAL header is used to store countersignature that provides authentic signing time. Only [RFC3161]([rfc3161](https://datatracker.ietf.org/doc/html/rfc3161#section-2.4.2)) compliant `TimeStampToken` are supported.
-  - **TODO** Define the opaque datum (hash of envelope) that is sent to TSA, and how TSA response (time stamp token) is represented in this header.
+- **`io.cncf.notary.timestampSignature`** (*bstr*): This OPTIONAL header is used to store a countersignature that proves the signature was generated before the time point been stamped. Only [RFC3161](https://datatracker.ietf.org/doc/html/rfc3161#section-2.4.2) compliant `TimeStampToken` are supported. If present, this header is validated and used solely under the [`notary.x509`](./signing-scheme.md/#notaryx509) signing scheme. Refer [*Timestamp Signature* unsigned attribute](signature-specification.md#unsigned-attributes) for more details.
```

Comment:
Why are we restring it to `notary.x509` and not including `notary.x509.signingAuthority`?

Reply:
@priteshbandi That's because for `notary.x509.signingAuthority`, we already have [Authentic Signing Time](https://github.com/notaryproject/specifications/blob/v1.0.0/specs/signature-specification.md#standard-attributes), it is a required field.
In our already released v1.0.0 spec, we have following: https://github.com/notaryproject/specifications/blob/v1.0.0/specs/signing-scheme.md#signature-creation
```
notary.x509.signingAuthority MUST use a timestamp attribute that is generated by a signing service as part of the original signature itself to determine authenticated signing time (timestamp). This is supported through the Authentic Signing time attribute in the signature envelope.
```
Therefore, including TSA signed timestamp token under `notary.x509.signingAuthority` would be a breaking change.

Reply:
@priteshbandi To answer the question in our meeting today, `Authentic Signing Time` in spec:
Authentic Signing Time (critical): The authenticated time at which the signature was generated. This claim allows a verifier to determine if the signature was generated when the certificate was valid.

So, by definition, this term is used to compare with certificate chain validity period. It's doing the same thing as the TSA Timestamp does. Therefore, from the spec point of view, we do not need another Timestamp field. 

However, as pointed out by @shizhMSFT, I think we need a revisit of the original design of `Authentic Signing Time`, because its trust source is still the signing authority itself (who also generates the Notation signature) instead of a third-party TSA. We have a couple of options:
1. Keep the current design for `notary.x509.signingAuthority`, no need for another TSA Timestamp field, no change to usage of `Authentic Signing Time`.
2. Split the trust source for `notary.x509.signingAuthority`, signing authority is only trusted for generating Notation signature. Timestamp still relies on TSA. In this case, the `Authentic Signing Time` can be removed as it duplicates with the `Signing Time` signed attribute.

What do you think?
@priteshbandi @shizhMSFT @yizha1 @FeynmanZhou

Reply:
Third option: For the signing scheme `notary.x509.signingAuthority`, if an `RFC-3161` timestamp is present, use it to validate the certificate chain, along with the 'Authentic Signing Time' checks. If it is not present, continue with the current behavior?

Reply:
> How about 3r option: For the signing scheme `notary.x509.signingAuthority`, if an `RFC-3161` timestamp is present, use it to validate the certificate chain, along with the 'Authentic Signing Time' checks. If it is not present, continue with the current behavior?

@priteshbandi SGTM. To confirm, for signing scheme `notary.x509.signingAuthority`, when both present, they all MUST pass the checks during verification right? (both of them are checking against the certificate chain).

Reply:
Resolving as discussed in the 3/25/2024 community meeting.